bucket/age.json

```
@@ -0,0 +1,15 @@
+{
+    "homepage": "https://age-encryption.org/",
+    "version": "1.0.0-beta2",
+    "description": "A simple, modern and secure encryption tool with small explicit keys, no config options, and UNIX-style composability.",
+    "license": {
+        "identifier": "BSD 3-Clause",
+        "url": "https://github.com/FiloSottile/age/blob/master/LICENSE"
+    },
+    "url": "https://github.com/FiloSottile/age/releases/download/v1.0.0-beta2/age-v1.0.0-beta2-windows-amd64.zip",
```

Since that we only have a 64-bit version of the app, we need to specify the `architecture` difference - otherwise our folks who use 32-bit Windows will be able to install the app, but not run it! (and that's weird)

We also need to specify the zip's checksum. You can get the checksum of a file with a tool like `hashcheck`, also available via scoop (`scoop bucket add extras` and `scoop install hashcheck`)

```suggestion
     "architecture": {
         "64bit": {
             "url": "https://github.com/FiloSottile/age/releases/download/v1.0.0-beta2/age-v1.0.0-beta2-windows-amd64.zip",
             "hash": "76c616ee56c2a2e182b50504d7a3b8ca390ee8bc2083a1dfd3cb7da50dcf70cc"
         }
     }
```

---

And autoupdate also need to be fixed.

---

Just checked that now, and it indeed doesn't work (because at the moment these are all prereleases)

---

It works. But you need to specify regex

---

In particular, we need to predict the wording they use. For the sake of simplicity, I'm thinking of `\/releases\/tag\/v([\\d.]+(-beta\\d+)?)` (The first part is needed, otherwise it'll catch garbage). Any way to shorten this even further?

---

... `tag\v([\\w.]+)`

---

Doesn't work because it doesn't match `-beta2` - so maybe `tag/v([\\w.]+(-[\\w]+)?)`?

---

Updated with de8a66f98ae4eb0252bf547257167221eadd7105

---

Forgot dash, `\w` contains only underscore

---

@linsui to correct `autoupdate` do i need to dupe the bin key inside of it? I'm not understanding the mistake based on the wiki.

---

I mean you need to use `architecture` in autoupdate. :smile: